ClimsoftVer4/ClimsoftVer4/formMetadata.vb

```diff
@@ -1836,6 +1835,53 @@ Err:
         txtClosingDate.Text = ClosingDate.Text
     End Sub
 
+    Private Function DMSToDD(Direction As Char, Deg As String, Min As String, Sec As String) As String
+        ' Convert value in Degrees, Minutes and Seconds (DMS) to Decimal Degrees (DD)
+        ' Direction must be N, S, E or W
+        If IsNumeric(Deg) And IsNumeric(Min) And IsNumeric(Sec) And Not Direction = vbNullChar Then
+            Dim multiplier As Integer = 1
+            Dim decimalDegrees As Double
+            If Direction = "S" OrElse Direction = "W" Then
+                multiplier = -1
+            End If
+            decimalDegrees = multiplier * (Val(Deg) + Val(Min) / 60 + Val(Sec) / 3600)
+            Return Math.Round(decimalDegrees, 2).ToString()
+        Else
+            Return ""
+        End If
+    End Function
+    Private Sub lstNS_Click(sender As Object, e As EventArgs) Handles lstNS.SelectedIndexChanged
+        txtLatitude.Text = DMSToDD(lstNS.SelectedItem, txtDegreesLat.Text, txtMinutesLat.Text, txtSecondsLat.Text)
+    End Sub
+
+    Private Sub lstEW_SelectedIndexChanged(sender As Object, e As EventArgs) Handles lstEW.SelectedIndexChanged
+        txtLongitude.Text = DMSToDD(lstEW.SelectedItem, txtDegreesLon.Text, txtMinutesLon.Text, txtSecondsLon.Text)
+    End Sub
+
+    Private Sub txtDegreesLat_TextChanged(sender As Object, e As EventArgs) Handles txtDegreesLat.TextChanged
```

Comment:
See all the `TextChanged` procedures below. These now update the Lat/Lon when any of the boxes change (not just if the Direction changes)